### PR TITLE
PEPPER-655 Just one RSAKeyProviderFactory, please

### PR DESCRIPTION
**Pull request by zyme**
While working through PEPPER-655, I started seeing some errors with DSM's requests to get jwks files (stack trace is below).  Although this PR does not directly address those findings, it should make it easier to address them.

We have two copies of RSAKeyProviderFactory, and what I've seeing is that DSS has no issue getting jwks files, but DSM does, so I'd like to consolidate the code into just one RSAKeyProviderFactory.  From there, I hope it will be easier to troubleshoot and get to root cause and a fix.

The main changes here are moving RSAKeyProviderFactory to ddp-common so both dss and dsm can use the same code, and centralizing the various auth0 dependencies.

```
java.lang.RuntimeException: Error reading jwk
	at org.broadinstitute.dsm.security.RSAKeyProviderFactory$1.getPublicKeyById(RSAKeyProviderFactory.java:26)
	at org.broadinstitute.dsm.security.RSAKeyProviderFactory$1.getPublicKeyById(RSAKeyProviderFactory.java:18)
	at com.auth0.jwt.algorithms.RSAAlgorithm.verify(RSAAlgorithm.java:41)
	at com.auth0.jwt.JWTVerifier.verify(JWTVerifier.java:354)
	at org.broadinstitute.dsm.security.Auth0Util.verifyAuth0Token(Auth0Util.java:191)
	at org.broadinstitute.dsm.util.JWTRouteFilter.isAccessAllowed(JWTRouteFilter.java:65)
	at org.broadinstitute.dsm.DSMServer.lambda$setupCustomRouting$7(DSMServer.java:665)
	at spark.FilterImpl$1.handle(FilterImpl.java:73)
	at spark.http.matching.BeforeFilters.execute(BeforeFilters.java:48)
	at spark.http.matching.MatcherFilter.doFilter(MatcherFilter.java:133)
	at org.broadinstitute.dsm.jetty.JettyConfig$JettyCustomRemoteAddrHeaderHandler.doHandle(JettyConfig.java:94)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1568)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:503)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:383)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:882)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1036)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.auth0.jwk.SigningKeyNotFoundException: Failed to get key with kid uRI5QmP3LzTAVN4TVjA51
	at com.auth0.jwk.GuavaCachedJwkProvider.get(GuavaCachedJwkProvider.java:53)
	at org.broadinstitute.dsm.security.RSAKeyProviderFactory$1.getPublicKeyById(RSAKeyProviderFactory.java:24)
	... 27 common frames omitted
Caused by: java.util.concurrent.ExecutionException: com.auth0.jwk.SigningKeyNotFoundException: Cannot obtain jwks from url https://dsm-dev.us.auth0.com/.well-known/jwks.json
	at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:592)
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:551)
	at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:111)
	at com.google.common.util.concurrent.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:247)
	at com.google.common.cache.LocalCache$Segment.getAndRecordStats(LocalCache.java:2345)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2313)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2189)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2079)
	at com.google.common.cache.LocalCache.get(LocalCache.java:4011)
	at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4919)
	at com.auth0.jwk.GuavaCachedJwkProvider.get(GuavaCachedJwkProvider.java:46)
	... 28 common frames omitted
Caused by: com.auth0.jwk.SigningKeyNotFoundException: Cannot obtain jwks from url https://dsm-dev.us.auth0.com/.well-known/jwks.json
	at com.auth0.jwk.UrlJwkProvider.getJwks(UrlJwkProvider.java:64)
	at com.auth0.jwk.UrlJwkProvider.getAll(UrlJwkProvider.java:71)
	at com.auth0.jwk.UrlJwkProvider.get(UrlJwkProvider.java:89)
	at com.auth0.jwk.RateLimitedJwkProvider.get(RateLimitedJwkProvider.java:30)
	at com.auth0.jwk.GuavaCachedJwkProvider$1.call(GuavaCachedJwkProvider.java:49)
	at com.auth0.jwk.GuavaCachedJwkProvider$1.call(GuavaCachedJwkProvider.java:46)
	at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4924)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3570)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2312)
	... 33 common frames omitted
Caused by: java.net.ConnectException: Connection timed out (Connection timed out)
	at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:412)
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:255)
	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:237)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.base/java.net.Socket.connect(Socket.java:609)
	at java.base/sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:305)
	at java.base/sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:173)
	at java.base/sun.net.NetworkClient.doConnect(NetworkClient.java:182)
	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:507)
	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:602)
	at java.base/sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:266)
	at java.base/sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:373)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:207)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1187)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1081)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:193)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1592)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1520)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:250)
	at java.base/java.net.URL.openStream(URL.java:1165)
	at com.auth0.jwk.UrlJwkProvider.getJwks(UrlJwkProvider.java:58)
	... 41 common frames omitted
```

### CODE REVIEW

File: pepper-apis/dss-core/pom.xml [hunk 1]
```diff
@@ -168,29 +168,6 @@
             <artifactId>bcpkix-jdk15on</artifactId>
         </dependency>
 
-        <dependency>
```
Comment (zyme) [inline pepper-apis/dss-core/pom.xml:171]: Moved to ddp-common's pom.xml

File: pepper-apis/ddp-common/pom.xml [hunk 2]
```diff
@@ -108,6 +108,27 @@
             <artifactId>sendgrid-java</artifactId>
             <version>4.8.1</version>
         </dependency>
+        <dependency>
```
Comment (zyme) [inline pepper-apis/ddp-common/pom.xml:111]: Moved from dss-core's pom.xml

File: pepper-apis/dsm-core/pom.xml [hunk 3]
```diff
@@ -95,17 +95,6 @@
             <artifactId>servo-core</artifactId>
         </dependency>
 
-        <!-- authorization and authentication-->
```
Comment (zyme) [inline pepper-apis/dsm-core/pom.xml:98]: Now these are coming in from ddp-common pom.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/security/RSAKeyProviderFactory.java [hunk 4]
```diff
@@ -1,42 +0,0 @@
-package org.broadinstitute.dsm.security;
```
Comment (zyme) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/security/RSAKeyProviderFactory.java:1]: This was a copy paste of DSS's RSAKeyProviderFactory

**Review by ssettipalli — APPROVED** [review]

**Pull request merged**